byterun/caml/misc.h

```diff
@@ -149,10 +149,10 @@ extern void caml_set_fields (intnat v, unsigned long, unsigned long);
 
 /* snprintf emulation for Win32 */
 
-#ifdef _WIN32
+#if defined(_WIN32) && defined(_MSC_VER) && _MSC_VER < 1900
```

Comment:
@alainfrisch is correct - unless you define `_ISOC99_SOURCE` or `__USE_MINGW_ANSI_STDIO`, mingw will map snprintf to Microsoft's non-C99 conformant `_snprintf`. The change is better predicated on the _UCRT define rather than the compiler version (although that's academic at the moment, as I think the Universal CRT can only be linked using Visual Studio 2015)